### PR TITLE
When unziping forms and mkdir use 0755

### LINKED ISSUE #281: Forms: updateforms permission denied
On a fresh install of [rmspi](https://github.com/toddejohnson/rmspi/releases/tag/v0.4) I run `pat updateforms` and get the below:

```sh
pi@rmspi:~ $ pat updateforms
2021/09/17 03:48:24 Updating form templates; current version is unknown
2021/09/17 03:48:25 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 03:48:28 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/IARU Forms/IARU Message Form.txt: permission denied
```

When I look at the folders:
```sh
pi@rmspi:~/.wl2k $ ls -ld Standard_Forms
drwxr-xr-x 3 pi pi    4096 Sep 17 03:48 Standard_Forms
pi@rmspi:~/.wl2k $ cd Standard_Forms/
pi@rmspi:~/.wl2k/Standard_Forms $ ls -l
total 4
drw-r--r-- 2 pi pi 4096 Sep 17 03:48 'IARU Forms'
pi@rmspi:~/.wl2k/Standard_Forms $ cd IARU\ Forms/
-bash: cd: IARU Forms/: Permission denied
```
So IARU Forms doesn't have 755 as a folder and it can't descend into the folder. 

```sh
pi@rmspi:~/.wl2k/Standard_Forms $ chmod 755 IARU\ Forms/
pi@rmspi:~/.wl2k/Standard_Forms $ cd IARU\ Forms/
pi@rmspi:~/.wl2k/Standard_Forms/IARU Forms $ cd ..
pi@rmspi:~/.wl2k/Standard_Forms $ pat updateforms
2021/09/17 03:59:06 Updating form templates; current version is unknown
2021/09/17 03:59:07 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 03:59:09 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/ICS USA Forms/10lineICS205_Initial.html: permission denied
```

So I work around it :

```sh
pi@rmspi:~/.wl2k/Standard_Forms $ for i in `seq 1 10`
do
find -type d -print0 | xargs -0 chmod 755
pat updateforms
done
```
Which runs through the cycle a few times:
```sh
2021/09/17 04:00:35 Updating form templates; current version is unknown
2021/09/17 04:00:36 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:00:38 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/OR STATE Forms/Oregon Activate Deactivate.txt: permission denied
2021/09/17 04:00:38 Updating form templates; current version is unknown
2021/09/17 04:00:39 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:00:41 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/OR STATE Forms/QUARTERLY_Test/Instructions.html: permission denied
2021/09/17 04:00:41 Updating form templates; current version is unknown
2021/09/17 04:00:42 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:00:44 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/RADIOGRAM & RRI Forms/Client Work Sheet.html: permission denied
2021/09/17 04:00:44 Updating form templates; current version is unknown
2021/09/17 04:00:45 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:00:50 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/SHARES Forms/SHARES Message Form2.txt: permission denied
2021/09/17 04:00:50 Updating form templates; current version is unknown
2021/09/17 04:00:51 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:00:53 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/TX STATE Forms/STAR_Form_TX_Initial.html: permission denied
2021/09/17 04:00:53 Updating form templates; current version is unknown
2021/09/17 04:00:54 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:00:57 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/USGS/USGS DYFI espanol.html: permission denied
2021/09/17 04:00:57 Updating form templates; current version is unknown
2021/09/17 04:00:58 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:01:00 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/WA STATE Forms/Emergency_Workers_Activity_WA_Initial.html: permission denied
2021/09/17 04:01:01 Updating form templates; current version is unknown
2021/09/17 04:01:01 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:01:03 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/WEATHER Forms/Hurricane Report Initial.html: permission denied
2021/09/17 04:01:04 Updating form templates; current version is unknown
2021/09/17 04:01:04 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/17 04:01:07 can't unzip forms update: open /home/pi/.wl2k/Standard_Forms/A - User Information/HTML Form Features.html: permission denied
2021/09/17 04:01:07 Updating form templates; current version is 1.0.164.0
2021/09/17 04:01:08 Latest forms version is 1.0.164; nothing to do
```

I think there is an issue in [unzip](https://github.com/la5nta/pat/blob/eb00df2b7509f62d0819d9b9189ee2f0218fa04e/internal/forms/forms.go#L378):
```go
		if f.FileInfo().IsDir() {
			os.MkdirAll(path, f.Mode())
```

I think this should read but have no clue what I'm doing:
```go
		if f.FileInfo().IsDir() {
			os.MkdirAll(path, 0755)
```

I'll file a PR but have no clue what I'm doing(and will say that there too)...

### PR DESCRIPTION
**Pull request by toddejohnson**
This fixes #281. I have no clue what this is doing though so revew is a must!

Before:
```sh
$ ./pat updateforms
2021/09/16 22:47:39 Updating form templates; current version is unknown
2021/09/16 22:47:40 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/16 22:47:42 can't unzip forms update: open /home/todd/.wl2k/Standard_Forms/IARU Forms/IARU Message Form.txt: permission denied
```
Clean Up:
```sh
$ rm ~/.wl2k/Standard_Forms/ -rf
$ ./make.bash
<<<SNIP>>>
Building Pat v0.11.0...
Enjoy!
```
After:
```sh
$ ./pat updateforms
2021/09/16 22:48:13 Updating form templates; current version is unknown
2021/09/16 22:48:13 Updating forms via https://api.onedrive.com/v1.0/shares/s!Ao2MJOj8BM8JomG20A77TzktEeyk/root/content?e=TA1LTz
2021/09/16 22:48:15 Finished forms update to 1.0.164
```

### CODE REVIEW

File: internal/forms/forms.go [hunk 1]
```diff
@@ -377,7 +377,7 @@ func unzip(src, dest string) error {
 		if f.FileInfo().IsDir() {
 			os.MkdirAll(path, f.Mode())
 		} else {
-			os.MkdirAll(filepath.Dir(path), f.Mode())
+			os.MkdirAll(filepath.Dir(path), 0755)
```
Comment (xylo04) [inline internal/forms/forms.go:380]: You can probably see above that this chunk of code was "heavily inspired" by StackOverflow. The point of f.Mode() is that the zip file has a preferred set of permissions it suggests to use, but I think you're right here, we can simply use 0755.

Can you please do the same 2 lines above for the directory case?

**Review by xylo04 — CHANGES_REQUESTED** [review]

**Comment by xylo04** [comment]
By the way, thanks for the PR!

File: internal/forms/forms.go (hunk 1 above)
Comment (toddejohnson) [reply internal/forms/forms.go:380]: lol! I wondered but in testing it didn't seem to affect the "issue" although like I said I have no clue. Change till it works right!

I've updated the branch with that as well. Better?

**Comment by toddejohnson** [comment]
I'm not sure why the build is failing now. This change I'd imagine is fairly trivial.

File: internal/forms/forms.go (hunk 1 above)
Comment (xylo04) [reply internal/forms/forms.go:380]: Looks good to me, thanks!

**Review by xylo04 — APPROVED** [review]

**Comment by toddejohnson** [comment]
I'm sorry I missed that the PR should be based on develop. I've made similar changes in [my develop fork](https://github.com/toddejohnson/pat/tree/develop) can't change that here. Should I close this PR and open a new to remove the conflict resolve step?

**Comment by martinhpedersen** [comment]
No worries! I can resolve the conflict when I'm merging 🙂 

Thank you! 👍 And thanks for reviewing, @xylo04.

**Comment by martinhpedersen** [comment]
Merged manually to `develop`.

**Pull request closed without merging**